pkg/common/common.go

```diff
@@ -71,6 +72,11 @@ const (
 	SpotConfigMapKey = "spot-deploy-config"
 )
 
+// Constant for node pool name validation
+const (
+	nodePoolNameValidationRegexp = "^[a-z0-9]*$"
```

Comment:
**Required**: this is more restrictive than it should be, we **MUST** relax it a bit to allow previously allowed names as well. 

According to the [AWS CloudFormation naming](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-name.html) (in the end we create CF stacks to manage most resources), dashes should definitely also be allowed, however they cannot begin or end the string or have 2 consecutive dashes.

Also according to my experience (supportive [link](https://gist.github.com/christianklotz/468a240a75f4ae3e73bd52722f926acb)) there is a character count limit on the AWS CF stack names (63 characters IIRC) which we should employ in order to prefilter the node pool names.
This limit should be lowered for the actual node pool name because we are stitching together the cluster name, node pool name and other IDs for the final node pool stack name, but it will have to suffice for now, at least we are guarding the most obvious mistakes.

Also it might be worth to use the [original Kubernetes DNS 1123 error message](https://github.com/kubernetes/kubernetes/pull/39675/files) or parts of it. This also states dots as allowed characters, I would give it a try whether they work just to make sure we don't overly limit the naming schema and also tailor the resulting error message accordingly.

P.S.: In the long run it would probably be a good idea to switch to using (short) unique IDs instead of names in stack naming, but that's a very long road to walk on.

Reply:
**Note/Recommended**: it is a good idea to include a comment reference next to the in-code raw regexes to some regex testing sites with saved examples (I often use regex101.com) to provide easy access to immediate verification with pre-saved test cases to see the regex in working. It is more prominent for complex regexes, but for newcomer's sake it is a good practice IMO.

Reply:
When I was checking the behavior with different node pool names, I encountered two different error messages (because the node pool name gets attached to the stack name and that has another constraints):

```
NodePoolLabelSet.labels.banzaicloud.io \"poolX\" is invalid: metadata.name: Invalid value: \"poolX\": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')
```
```
ValidationError: 1 validation error detected: Value 'pipeline-eks-nodepool-test-validation-pool1X'' at 'stackName' failed to satisfy constraint: Member must satisfy regular expression pattern: [a-zA-Z][-a-zA-Z0-9]*
```

So it seemed to me that is why the wording in the [issue](https://github.com/banzaicloud/pipeline/issues/2622) says "must consist of lower case alphanumeric characters", to satisfy both of these regexes. You are right about the `-` though and the character limit is also a good idea, I am going to refine the regexp!

Reply:
I crafted a regexp that allows single dashes within the string and also added node pool name verification.

Reply:
Then you are right, the correct approach is to use the top regex + the length constraint.